process/accountProcessor.go

```diff
@@ -502,13 +502,18 @@ func (ap *AccountProcessor) getShardIfOdAddress(address string) (uint32, error)
 	return ap.proc.ComputeShardId(addressBytes)
 }
 
-func (ap *AccountProcessor) getObserversForAddress(address string, availability data.ObserverDataAvailabilityType) ([]*data.NodeData, error) {
+func (ap *AccountProcessor) getObserversForAddress(address string, availability data.ObserverDataAvailabilityType, forcedShardID core.OptionalUint32) ([]*data.NodeData, error) {
 	addressBytes, err := ap.pubKeyConverter.Decode(address)
 	if err != nil {
 		return nil, err
 	}
 
-	shardID, err := ap.proc.ComputeShardId(addressBytes)
+	var shardID uint32
+	if forcedShardID.HasValue {
+		shardID = forcedShardID.Value
+	} else {
+		shardID, err = ap.proc.ComputeShardId(addressBytes)
```

Comment:
addressBytes can be computed here. in case of a forced shard id, there's no need to compute it

Reply:
fixed